server/server.js

```diff
@@ -172,6 +190,106 @@ app.post('/genotet/upload', upload.single('file'), function(req, res) {
   });
 });
 
+app.use(cookieParser());
+
+app.post('/genotet/user', function(req, res) {
+  console.log('POST user');
+
+  var type = req.body.type;
+  var userInfo = {};
+  var response;
+  switch (type) {
+    case user.QueryType.SIGHUP:
+      data = {
+        email: req.body.email,
+        username: req.body.username,
+        password: req.body.password,
+        confirmed: req.body.confirmed
+      };
+      response = user.signUp(userPath, data);
+      break;
+    case user.QueryType.SIGNIN:
+      data = {
+        username: req.body.username,
+        password: req.body.password
+      };
+      response = user.signIn(userPath, data);
+      break;
+
+    // Undefined type, error
+    default:
+      console.error('invalid query type');
+      data = {
+        error: {
+          type: 'query',
+          message: 'invalid query type'
+        }
+      };
+  }
+  res.header('Access-Control-Allow-Origin', '*');
+  if (data.error) {
+    console.log(data.error);
+    res.status(500).json(data.error);
+  } else if (response.success) {
```

Comment:
If username/email are not unique, or username and password are not inconsisant, the program shall return error response rather than an object with success field.